tests/integration/dynamic_ip_test.py

```diff
@@ -271,10 +265,9 @@ def test_ipv4_dhcp_ignore_gateway(dhcpcli_up):
     libnmstate.apply(desired_state)
 
     assertlib.assert_state(desired_state)
-    time.sleep(5)  # wait to get resolv.conf updated
     assert _has_ipv4_dhcp_nameserver()
-    assert not _has_ipv4_dhcp_gateway()
     assert _has_ipv4_classless_route()
+    assert not _has_ipv4_dhcp_gateway(False)
```

Comment:
Why does it sometimes use polling and sometimes not? Do you disable polling for negated checks? Wont't this lead to false positives if the gateway will appear shortly after this check? I guess for these negated checks, Nmstate would have to check that the respective NM profiles are in a final state (not waiting for any more DHCP responses) and do the check then. Not sure if this is possible atm, though.

Thinking this further, actually `libnmstate.apply() should block or allow to block until the state is as expected.

But I guess for the current code, IMHO it would make more sense to have `_has_ipv4_dhcp_gateway()` not do the polling but have a poll wrapper that will be used if necessary and then the negated check can be done without the poll ing for now.

Reply:
If previous check returned, that means DHCP has finished its work. There is no need to waste 20 more seconds.